main/model.h

```diff
@@ -6,11 +6,22 @@
 #include <memory>
 #include <vector>
 
-class Model {
+class ModelInterface {
+public:
+  virtual ~ModelInterface(){};
+  // Model that simulation will be performend on
+  virtual std::vector<objects::TriangleObj *> triangles() const = 0;
+  virtual float height() const = 0;
+  virtual float sideSize() const = 0;
+  virtual bool empty() const {
+    return std::max<float>(height(), sideSize()) <= constants::kAccuracy;
```

Comment:
nit: is <float> needed here?

Reply:
its not. I forgot to revert this. When I was checking out gMock. fixed